chore: add more tests for MarkdownPod

hikchoi opened the pull request:
- Added additional tests for `MarkPublishPod`
Closes https://github.com/dendronhq/dendron/issues/417

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dendronhq/dendron?pullRequest=458) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dendronhq/dendron?pullRequest=458) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dendronhq/dendron?pullRequest=458) it.</sub>

kevinslin commented on packages/pods-core/src/builtin/__tests__/MarkdownPod.spec.ts:
```diff
@@ -78,4 +78,129 @@ describe("MarkdownPublishPod", () => {
+    );
+  });
+
+  test("journal", async () => {
```
since this case isn't really testing anything `setupJournals` was a preset to test that sibling navigation worked, we don't need to include this as a test case

kevinslin commented on packages/pods-core/src/builtin/__tests__/MarkdownPod.spec.ts:
```diff
@@ -78,4 +78,129 @@ describe("MarkdownPublishPod", () => {
     );
   });
+
+  test("schema with template", async () => {
```
`schemaPreset` was to test that template schemas worked. I would use these tests more to test special cases for markdown (eg. what happens when we publish an empty note? a note with a wiki link? a note with a relative link? etc)

hikchoi replied:
Will think about some edge cases then 👍

kevinslin commented on packages/pods-core/src/builtin/__tests__/MarkdownPod.spec.ts:
```diff
@@ -78,4 +78,129 @@ describe("MarkdownPublishPod", () => {
+    );
+  });
+
+  test("schema with namespace template", async () => {
```
same comment as above. this tests for schema functionality but doesn't test something specific about Publish

kevinslin reviewed (changes requested):
see comments

hikchoi commented:
I've removed the unnecessary tests and added tests for dendron specific markdowns (wikilinks, various refs). 
I tried to cover everything I could find that are dendron specific markdown, but I might have missed some things, so please let me know if that's the case.

Also bundled related cases into one test block, but I'm not sure if that's how you would usually lay out your tests.
If it's better to have them separately, I'll do that.

kevinslin commented:
awesome, thanks for the changes. just merged :)
